src/main/java/com/minecolonies/coremod/client/gui/WindowBuildTool.java

```diff
@@ -2,55 +2,84 @@
 
 import com.minecolonies.blockout.Log;
 import com.minecolonies.blockout.controls.Button;
+import com.minecolonies.blockout.views.DropDownList;
 import com.minecolonies.coremod.MineColonies;
+import com.minecolonies.coremod.colony.ColonyManager;
 import com.minecolonies.coremod.colony.Structures;
 import com.minecolonies.coremod.lib.Constants;
 import com.minecolonies.coremod.network.messages.BuildToolPlaceMessage;
+import com.minecolonies.coremod.network.messages.SchematicRequestMessage;
+import com.minecolonies.coremod.network.messages.SchematicSaveMessage;
 import com.minecolonies.coremod.util.BlockUtils;
 import com.minecolonies.coremod.util.LanguageHandler;
 import com.minecolonies.structures.helpers.Settings;
 import com.minecolonies.structures.helpers.Structure;
 import net.minecraft.block.Block;
-import net.minecraft.client.Minecraft;
 import net.minecraft.entity.player.InventoryPlayer;
 import net.minecraft.item.ItemStack;
 import net.minecraft.util.Rotation;
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.gen.structure.template.PlacementSettings;
+import net.minecraftforge.fml.common.FMLCommonHandler;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
-import java.io.File;
-import java.io.IOException;
+import java.io.InputStream;
 import java.util.ArrayList;
-import java.util.Collections;
 import java.util.List;
-import java.util.stream.Collectors;
+
+import java.util.function.ObjIntConsumer;
+
 
 /**
  * BuildTool window.
  */
-public class WindowBuildTool extends AbstractWindowSkeleton
+public class WindowBuildTool extends AbstractWindowSkeleton implements DialogDoneCancel.Handler, ObjIntConsumer<DropDownList>
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This class has 6 parents which is greater than 5 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)